### PR TITLE
Invert main strategy loop for performance and customizability

### PR DESCRIPTION
**Pull request by damemi**
I noticed this while thinking about the new API options we started discussing in https://github.com/kubernetes-sigs/descheduler/issues/486

Right now we are checking against every strategy function and then seeing if it's enabled. This does 2 things:
1. Wastes time for strategies that aren't enabled
2. Only allows strategies to be run in a fixed order

1. is probably negligible, but 2. is something we should consider providing. I don't know if the current order is hardcoded for a reason. There also hasn't been any demand for this specific functionality that I know of, but while we are considering improvements to the API I think a change like this could make it more flexible.

Either way, this does also add an error for invalid strategy names (where it currently just silently skips them)

### CODE REVIEW

File: pkg/descheduler/descheduler.go [hunk 1]
```diff
@@ -136,9 +136,11 @@ func RunDeschedulerStrategies(ctx context.Context, rs *options.DeschedulerServer
 			ignorePvcPods,
 		)
 
-		for name, f := range strategyFuncs {
-			if strategy := deschedulerPolicy.Strategies[api.StrategyName(name)]; strategy.Enabled {
+		for name, strategy := range deschedulerPolicy.Strategies {
+			if f := strategyFuncs[name]; f != nil && strategy.Enabled {
 				f(ctx, rs.Client, strategy, nodes, podEvictor)
+			} else {
+				klog.Errorf("Unknown strategy name '%s', exiting", name)
```
Comment (ingvagabund) [inline pkg/descheduler/descheduler.go:143]: This will log even when the strategy exists but is not enabled.

Comment (damemi) [reply pkg/descheduler/descheduler.go:145]: Good catch, thanks

File: pkg/descheduler/descheduler.go [hunk 2]
```diff
@@ -136,9 +136,13 @@ func RunDeschedulerStrategies(ctx context.Context, rs *options.DeschedulerServer
...
-		for name, f := range strategyFuncs {
-			if strategy := deschedulerPolicy.Strategies[api.StrategyName(name)]; strategy.Enabled {
-				f(ctx, rs.Client, strategy, nodes, podEvictor)
+		for name, strategy := range deschedulerPolicy.Strategies {
+			if f, ok := strategyFuncs[name]; ok {
+				if strategy.Enabled {
+					f(ctx, rs.Client, strategy, nodes, podEvictor)
+				}
+			} else {
+				klog.Errorf("Unknown strategy name '%s', exiting", name)
```
Comment (ingvagabund) [inline pkg/descheduler/descheduler.go:145]: Maybe skipping instead of existing? Exiting might give impression the descheduler is aborting actually.

**Pull request merged**